### PR TITLE
Code Improvements

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Code improvements, fine-tuning, and a fix.

### CODE REVIEW

**Comment by Martinski4GitHub** [comment]
@ExtremeFiretop,

OK. I'm done with my changes. Please review as needed.
BTW, the fix was to remove the remaining call to the "`_WAN_IsConnected_`" function which no longer exists.

I have nothing else to add for this release, so whenever you're ready feel free to issue the production release.

We had a very good, productive weekend, I'd say. :>).

File: MerlinAU.sh [hunk 1]
```diff
@@ -6462,14 +6467,16 @@ _PostRebootRunNow_()
    #--------------------------------------------------------------
    while [ "$curWaitDelaySecs" -lt "$maxWaitDelaySecs" ]
    do
-      if _WAN_IsConnected_ && \
-         [ -d "$FW_ZIP_BASE_DIR" ] && \
+      if [ -d "$FW_ZIP_BASE_DIR" ] && \
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:6470]: Lool! I didn't even notice this was still here

**Review by ExtremeFiretop — APPROVED** [review]

**Comment by ExtremeFiretop** [comment]
Approved and good to go!

**Comment by ExtremeFiretop** [comment]
> I have nothing else to add for this release, so whenever you're ready, feel free to issue the production release.
> 
Currently playing some Minecraft with the Girlfriend but I'll issue a production releases in the next half hour or so, I have to finish building this stone house and then I'll start updating the forms post, etc! :)

> We had a very good, productive weekend, I'd say. :>).
> 

Agreed! Some good troubleshooting on some tricky issues. Probably would of gone unreported by most for a while but we nailed it down in the end!

> P.S. BTW, I ran this latest script version through my company's Linter tool: no errors found and no new warnings at all. Very clean!! :>)

Just what I like to hear! 😌 I mean, did it take a bit of guess work and testing to get it all down, yes, but we got there! And that's always what matters in the end.

**Comment by Martinski4GitHub** [comment]
> > I have nothing else to add for this release, so whenever you're ready, feel free to issue the production release.
> 
> Currently playing some Minecraft with the Girlfriend but I'll issue a production releases in the next half hour or so, I have to finish building this stone house and then I'll start updating the forms post, etc! :)
> 
>
No worries, bud. Enjoy your time with your girlfriend. Those moments are a part of what keeps a relationship healthy & happy.

>
> > We had a very good, productive weekend, I'd say. :>).
> 
> Agreed! Some good troubleshooting on some tricky issues. Probably would of gone unreported by most for a while but we nailed it down in the end!
> 
>
Yeah, those issues that happen during reboot can be difficult to catch & debug because so many other processes are starting up at the same time, plus whatever other add-ons the user may have installed, and any interaction between them can change the dynamics, the flow, and the timing of the boot sequence. But, yeah, I think we nailed this one.

>
> > P.S. BTW, I ran this latest script version through my company's Linter tool: no errors found and no new warnings at all. Very clean!! :>)
> 
> Just what I like to hear! 😌 I mean, did it take a bit of guess work and testing to get it all down, yes, but we got there! And that's always what matters in the end.
>
>
Not to brag, but our shell script has the cleanest code of all the other scripts I have seen/reviewed from the SNB Forums, by far.

**Comment by ExtremeFiretop** [comment]
> No worries, bud. Enjoy your time with your girlfriend. Those moments are a part of what keeps a relationship healthy & happy.
> 

I'm done with the stone house haha!
Working on the release notes and updating the readme now! :)

> Yeah, those issues that happen during reboot can be difficult to catch & debug because so many other processes are starting up at the same time, plus whatever other add-ons the user may have installed, and any interaction between them can change the dynamics, the flow, and the timing of the boot sequence. But, yeah, I think we nailed this one.
> 

I've be VERY surprised if someone said they still had issues; we nailed it down good ;) More like screwed it down through 3 boards. It ain't going nowhere. ;)

> Not to brag, but our shell script has the cleanest code of all the other scripts I have seen/reviewed from the SNB Forums, by far.

Why do you think I only ever want to work on our script? LOL I've looked at many of the other scripts, maybe it's just because I've so involved with this one, but it takes so much more work to decode whats going on compares to ours. Issue comes up? Following the code here is like music to my ears how clean it is!

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

If I could add anymore positive reactions to that last post, I would!

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> If I could add anymore positive reactions to that last post, I would!
>
👍👍👍
 LOL!!!

**Comment by Martinski4GitHub** [comment]
> > No worries, bud. Enjoy your time with your girlfriend. Those moments are a part of what keeps a relationship healthy & happy.
> 
> I'm done with the stone house haha! Working on the release notes and updating the readme now! :)
> 
> > Yeah, those issues that happen during reboot can be difficult to catch & debug because so many other processes are starting up at the same time, plus whatever other add-ons the user may have installed, and any interaction between them can change the dynamics, the flow, and the timing of the boot sequence. But, yeah, I think we nailed this one.
> 
> I've be VERY surprised if someone said they still had issues; we nailed it down good ;) More like screwed it down through 3 boards. It ain't going nowhere. ;)
>
>
Yep, it's screwed in & nailed tight - with 10-inch nails & screws :>) LOL.

> 
> > Not to brag, but our shell script has the cleanest code of all the other scripts I have seen/reviewed from the SNB Forums, by far.
> 
> Why do you think I only ever want to work on our script? LOL I've looked at many of the other scripts, maybe it's just because I've so involved with this one, but it takes so much more work to decode whats going on compares to ours. Issue comes up? Following the code here is like music to my ears how clean it is!
>
>
Yeah, I hear you. IMO, the 2nd cleanest scripts that I've seen are from Jack Yaz. It's not a close 2nd, but his scripts are fairly clean so it didn't take me long before I understood his code and the logic. Everyone else's that I've seen are many miles away from that which makes them more difficult to work with.

**Pull request merged**